ext/msgpack/buffer.h

```diff
@@ -485,7 +483,7 @@ static inline VALUE msgpack_buffer_read_top_as_string(msgpack_buffer_t* b, size_
 #endif //STR_UMINUS_DEDUPE_FROZEN
         // MRI 2.5 and older do not deduplicate strings that are already
         // frozen.
-        result = rb_funcall(result, s_uminus, 0);
+        result = rb_funcall(result, rb_intern("-@"), 0);
```

Comment:
Does this change mean that unpacking frozen strings will call `#-@` every time?

Reply:
It looks not a bad idea to create the symbol in the same way with #253
WDYT?

Reply:
That's what this PR is about. 

Some functions like `rb_funcall` don't take a Symbol (`VALUE`), they take an `ID` which can be created statically through `rb_intern()`, and Ruby's macro take care of it.

If you look at MRI's source, you'll see that it rarely bother pre-initializing `ID`.

But it's not like it's hurting either, if you like it that way, it's fine I can close.